core/server/master/src/main/java/alluxio/master/metastore/caching/Cache.java

```diff
@@ -85,8 +90,17 @@ public Cache(CacheConfiguration conf, String name, MetricKey metricKey) {
     mEvictionThread = new EvictionThread();
     mEvictionThread.setDaemon(true);
     // The eviction thread is started lazily when we first reach the high water mark.
+    mStatsCounter = new StatsCounter();
 
     MetricsSystem.registerGaugeIfAbsent(metricKey.getName(), mMap::size);
+    MetricsSystem.registerGaugeIfAbsent(metricKey.getName() + "-hits",
```

Comment:
Can you help change to use a different MetricKey for each Metric?
In this way, we have a central place storing all the metric names and their descriptions.